### PR TITLE
Update Maespa interface scripts to use "Maeswrap" package

### LINKED ISSUE #733: Add Maespa Model
This issue will correspond with the work being done to couple the Maespa Model http://maespa.github.io/ to PEcAn.

### LINKED ISSUE #817: MAESPA output units
In model2netcdf.MAESPA.R the flux units are currently being returned as, for example, gross photosynthesis mol tree-1 d-1. These units are bit useless if you ever wish to compare the fluxes to any other models or data.

To get them in sensible m-2 units you need to:
e.g. for GPP (python pandas syntax)
``` python
df.loc[:,"totPs"] *= MOLE_C_TO_GRAMS_C * num_trees / plot_size
```
where plot size is xmax * ymax. The information on plot_size and num_trees is in the input file trees.dat

This is therefore another instance where one would need to be able to access the input file. If this is going to be a problem I can speak to Remko about potentially changing this, though I imagine he won't be keen as this may break a lot of their existing structure.

### PR DESCRIPTION
**Pull request by tonygardella**
Changed met2model and model2ncdf so that they use Maeswrap R package. Will put instructions of how to install in documentation. It has a dependency package "rgl" that gives errors when using the regular install.packages function. Also, added to write.configs to handle the rest of the input files and write out jobs.sh. Roxygen2 and formatR were used for ensuring package formatting and structure.

This should fix #817.

To close #733, input/output files need to be checked for accuracy.

@mdietze is up for review.

### CODE REVIEW

File: models/maespa/R/met2model.MAESPA.R [hunk 1]
```diff
@@ -109,7 +109,7 @@ met2model.MAESPA <- function(in.path, in.prefix, outfolder, start_date,
      RAD <- ncvar_get(nc,"surface_downwelling_shortwave_flux_in_air") #W m-2
      PAR <- try(ncvar_get(nc,"surface_downwelling_photosynthetic_photon_flux_in_air")) #mol m-2 s-1
      TAIR <- ncvar_get(nc,"air_temperature") # K
-     QAIR <- ncvar_get(nc,"specific_humidity"))# 1
+     QAIR <- ncvar_get(nc,"specific_humidity")# 1
      PPT <- ncvar_get(nc,"precipitation_flux") #kg m-2 s-1
      CA <- try(ncvar_get(nc,"mole_fraction_of_carbon_dioxide_in_air")) #mol/mol
      PRESS <- ncvar_get(nc,"air_pressure")# Pa
```
Comment (bpbond) [inline models/maespa/R/met2model.MAESPA.R]: Why are some `ncvar_get()` calls wrapped in `try()` but most not? Are these variables that might not be in the netcdf files, while the others are guaranteed to be present?

Comment (mdietze) [inline models/maespa/R/met2model.MAESPA.R]: Yep, exactly

Comment (tonygardella) [inline models/maespa/R/met2model.MAESPA.R]: Yes!

File: models/maespa/R/met2model.MAESPA.R [hunk 2]
```diff
@@ -29,172 +29,180 @@
...
+     
+     ####ppm. atmospheric CO2 concentration. Constant from Enviiron namelist used instead
+     if (!is.numeric(CA)) {
+       print(
+         "Atmospheric CO2 concentration will be set to constant value set in ENVIRON namelist "
+       )
+       rm(CA)
+       defaultCO2 = 400 #400 is estimation of atmospheric CO2 in ppm)
+     } else {
+       defaultCO2 = 400
```
Comment (mdietze) [inline models/maespa/R/met2model.MAESPA.R]: I'm confused here and line 198 -- it appears you're using the default CO2 value even if CO2 data is provided.

File: models/maespa/R/met2model.MAESPA.R [hunk 3]
```diff
@@ -29,172 +29,180 @@
...
+   
+ }### end loop over years
+ 
+ 
+ #Get names for columns of variable table
+ columnnames <-  paste0(rownames(tmp),collapse = "'     '")
+ #Get number of variables
+ numbercolumns <- nrow(out)
+ #turn into matrix
+ out <- matrix(out,ncol = numbercolumns, byrow=TRUE)
```
Comment (mdietze) [inline models/maespa/R/met2model.MAESPA.R]: Why are you setting the number of columns to the number of rows?  If out needs to be a matrix, wouldn't as.matrix be simpler?

Comment (tonygardella) [inline models/maespa/R/met2model.MAESPA.R]: I'll go over this to make it cleaner.

**Pull request merged**